### PR TITLE
client/WIP: Add tests

### PR DESCRIPTION
**Pull request by sarahhodne**
Work in progress, please do not merge yet.

Submitting this to see what you think of the testing. Should I test more/less? Does the test style seem OK?

I'll add more tests tomorrow and write a proper commit message.

Also, note the FIXME. I didn't want to change implementation code yet, but I think maybe the endpoint should be a member on the `Client` struct.

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    err := json.NewDecoder(req.Body).Decode(&payload)
+    if err != nil {
+        fmt.Println("Got error:", err)
+    }
+
+    return
+}
+
+func stringSlicesEqual(a, b []string) bool {
```
Comment (sarahhodne) [inline client_test.go:38]: Note to self: This function is no longer being used, remove.

Comment (joshk) [inline client_test.go:38]: So why not remove it? :)

File: client_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "net/url"
+    "reflect"
+    "testing"
+)
+
+func setupTestServer(handler http.Handler) (server *httptest.Server) {
+    server = httptest.NewServer(handler)
+    url, _ := url.Parse(server.URL)
+    // FIXME: This is not thread-safe at all, and will not handle running tests in parallel.
+    //   The endpoint field should probably be moved to the Client struct.
```
Comment (joshk) [inline client_test.go:17]: I think this is fine as this way of setting up a test server is recommended.

**Comment by joshk** [comment]
I think this looks like a great start, you rock Henrik!

File: client_test.go (hunk 2 above)
Comment (sarahhodne) [inline client_test.go:17]: Setting up the test server is fine, it's the global variable that is setting off my code smell alarm.

Comment (joshk) [inline client_test.go:17]: which global var?

Comment (sarahhodne) [inline client_test.go:17]: `pusher.Endpoint`

Comment (joshk) [inline client_test.go:17]: ahhhhh, ok, i get cha, well, that is a default, and yeah, i am not sure if it is bad or not.

**Pull request merged**